api/class-wc-rest-dev-data-counts-controller.php

```diff
@@ -0,0 +1,253 @@
+<?php
+/**
+ * REST API Data object counts controller.
+ *
+ * Handles requests to the /data/counts endpoint.
+ *
+ * @author   Automattic
+ * @category API
+ * @package  WooCommerce/API
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * REST API Data Counts controller class.
+ *
+ * @package WooCommerce/API
+ */
+class WC_REST_Dev_Data_Counts_Controller extends WC_REST_Dev_Data_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'data/counts';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Return the list of order counts.
+	 *
+	 * @return array list of counts for all order statuses
+	 */
+	public function get_order_counts() {
+		$statuses = array_keys( wc_get_order_statuses() );
+		// Pre-fill the array with all statuses at zero
+		$counts = array_combine( $statuses, array_fill( 0, count( $statuses ), 0 ) );
+		foreach ( wc_get_order_types( 'order-count' ) as $type ) {
+			$counts_for_type = (array) wp_count_posts( $type );
+			foreach ( $statuses as $status ) {
+				$counts[ $status ] += $counts_for_type[ $status ] ?: 0;
+			}
+		}
+		return $counts;
+	}
+
+	/**
+	 * Return the list of product counts.
+	 *
+	 * @return array list of counts for all product statuses
+	 */
+	public function get_product_counts() {
+		global $wpdb;
+		$count = get_transient( 'wc-counts-products' );
+		if ( false !== $count ) {
+			return $count;
+		}
+
+		$args = array(
+			'status' => 'publish',
+			'limit' => -1,
```

Comment:
I'm not thrilled about the unlimited query, but I'm not sure if there's another option here? I want to get all products, then sort by inventory level.